ROADMAP.md

```diff
@@ -0,0 +1,54 @@
+# 2024 Roadmap
+
+The conda-store roadmap for the remaining part of 2024 into 2025 will revolve around several key areas of iterative improvement:
+
+- Performance
+- Backup and restore
+- User experience improvements
+
+| Mark | Description      |
+| ---- | ---------------- |
+| 📪   | work not started |
+| ✅   | work completed   |
+| ⬆️   | upstream         |
+| 🏃   | in progress      |
+| 💪   | stretch goal     |
+
+
+## Performance
+- ✅ Identify the parts of conda-store responsible for slowing down new environment builds.
+- 🏃 ~40% of environment build time comes from a slow call to conda-lock, which solves each environment twice (once for conda packages, and again to support pip packages with a vendored version of poetry). The dependency on conda-lock is to be eliminated by implementing official conda support for lockfile generation.
+- ⬆️🏃 Teach conda create  how to handle environment.yml files (#14113)
+- ⬆️🏃 Add support for conda [env] hooks (#13833).
+- ⬆️📪 Implement the aforementioned hooks in conda-pypi
+- ⬆️📪 Define and standardize a conda lockfile schema.
+- ⬆️📪 Replace conda-lock with an officially supported conda plugin for lockfile generation.
+- 🏃 ~40% of environment build time is due to package download and extraction, which is carried out serially to avoid collisions when multiple concurrent requests for the same package are made. Implement a locking mechanism to allow concurrent downloads, reducing this to just the bare network time required to download the required assets.
+- 🏃 Currently, simultaneous calls to conda create result in corrupted packages and cached repo data (#13037). Support for locking the repodata cache was added by the community recently (#12996), but full locking support must be added to prevent conflicts/data corruption from other concurrent conda operations.
+- ⬆️📪 Add native conda support for locking to allow parallel conda commands on a single machine (#13055).
+
+## Storage / Backup and Restore
+
+- 📪 A major concern for existing conda-store deployments is that old, unused builds still take up space on shared volumes. Old builds eventually fill up disk space for deployments with many users, requiring manual intervention to delete them. Implementing a backup mechanism will require reaching consensus on an approach to archiving and recreation old builds (#889).
+- 📪 Implement archiving of old builds: the yaml spec and the lockfile are kept, but the environments on disk are deleted.
+- 📪 Implement bulk archiving of old builds. --all-but-last, --older-than=<date>.
```

Comment:
The `--all-but-last, --older-than=<date>` seem out of place, are these CLI flags only then? Are they meant to be archiving approaches/methods?

Reply:
These were meant to be CLI options, but I think this is an implementation detail so I've just removed them. We may want to implement a UI component that does this as well, so let's just leave these out.